src/fromager/commands/bootstrap.py

```diff
@@ -189,64 +201,102 @@ def write_constraints_file(
+    resolved_something = True
+
+    # Outer while loop to resolve remaining dependencies with multiple versions
+    while unresolved_dependencies and resolved_something:
+        resolved_something = False
+        # Make copy of the original list
+        items_to_resolve = unresolved_dependencies[:]
+        # Loop over items_to_resolve. It is a tuple with dep name and a list of edges
+        for dep_name, nodes in items_to_resolve:
+            # Track which versions can be used by which parent requirement.
+            usable_versions: dict[str, list[Version]] = {}
+            # Track how many total users of a requirement (by name) there are so we
+            # can tell later if any version can be used by all of them.
+            user_counter = 0
+            # Which parent requirements can use which versions of the dependency we
+            # are working on?
+            dep_versions = [node.version for node in nodes]
+            # Loop over the nodes list
+            for node in nodes:
+                parent_edges = node.get_incoming_install_edges()
+                # Loop over parent_edges list
+                for parent_edge in parent_edges:
+                    parent_name = parent_edge.destination_node.canonicalized_name
+                    # Condition to select the right version.
+                    # We check whether parent_name is already in resolved dict and the version associated with that
+                    # is not the version of the destination node
+                    if (
+                        parent_name in resolved
+                        and resolved[parent_name]
+                        != parent_edge.destination_node.version
+                    ):
+                        continue
+                    # Loop to find the usable versions
+                    for matching_version in parent_edge.req.specifier.filter(
+                        dep_versions
+                    ):
+                        usable_versions.setdefault(str(matching_version), []).append(
+                            parent_edge.destination_node.version
+                        )
+                    user_counter += 1
+
+            # Look for one version that can be used by all the parent dependencies
+            # and output that if we find it. Otherwise, include a warning and report
+            # all versions so a human reading the file can make their own decision
+            # about how to resolve the conflict.
+            for v, users in usable_versions.items():
+                if len(users) == user_counter:
+                    version_strs = [str(v) for v in sorted(versions)]
+                    logger.debug(
+                        "%s: selecting %s from multiple candidates %s",
+                        dep_name,
+                        v,
+                        version_strs,
                     )
+                    # Set the flag to true since we were able to resolve something
```

Comment:
This comment is 2 lines away from the actual code. I think we can just delete this line to clean it up.

Reply:
okay, done